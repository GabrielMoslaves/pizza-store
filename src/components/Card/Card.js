import React from "react";
import styles from "./styles.module.scss";
import Button from "../Button/Button";

const Card = (props) => {
  const { name, description, price, image, product, setSelected } = props;

  return (
    <div className={styles.container}>
      <img src={image} alt="pizza" />
      <h3>{name}</h3>
      <p>{description}</p>
      <h1>R${price}</h1>
      <Button
        text="Comprar agora"
        onClick={() => setSelected((prevState) => [...prevState, product])}
      />
    </div>
  );
};

export default Card;
