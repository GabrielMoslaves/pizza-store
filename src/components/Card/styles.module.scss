.container {
  background-color: red;
  display: flex;
  width: 256px;
  height: 275px;
  border-radius: 25px;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  background-color: #eed7d2;
  text-align: center;
  h1 {
    margin: 15px;
  }
}
