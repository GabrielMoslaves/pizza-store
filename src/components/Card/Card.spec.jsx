import { fireEvent, render, screen } from "@testing-library/react";
import Card from "./Card";

describe("<Card/>", () => {
  const mockCardProps = {
    name: "Pizza",
    description: "lorem ipsum dolor",
    price: 30,
    image: "./pizza1.png",
  };

  it("should render card correctly", () => {
    render(<Card {...mockCardProps} />);

    expect(screen.getByRole("img", { name: /pizza/i })).toHaveAttribute("src", "./pizza1.png");
    expect(screen.getByRole("heading", { name: /pizza/i })).toBeInTheDocument();
    expect(screen.getByText(/lorem ipsum dolor/i)).toBeInTheDocument();
    expect(screen.getByRole("heading", { name: /30/i })).toBeInTheDocument();
    expect(screen.getByRole("button", { name: /comprar agora/i })).toBeInTheDocument();
  });

  it("should match snapshot", () => {
    const { container } = render(<Card {...mockCardProps} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
