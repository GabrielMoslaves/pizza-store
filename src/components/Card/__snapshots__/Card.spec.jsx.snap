// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card/> should match snapshot 1`] = `
<div
  class="container"
>
  <img
    alt="Pizza"
    src="./pizza1.png"
  />
  <h3>
    Pizza
  </h3>
  <p>
    lorem ipsum dolor
  </p>
  <h1>
    R$
    30
  </h1>
  <button
    class="container"
  >
    Comprar agora
  </button>
</div>
`;
